lib/path/tests/normalise.nix

```
@@ -0,0 +1,34 @@
+# Used by ./prop.sh
+#
+# stringsDir is a flat directory containing files with randomly-generated
+# path-like values
+#
+# This file should return a { <value> = <lib.path.subpath.normalise value>; }
+# attribute set for each value. If `normalise` fails to evaluate,
+# "" is returned instead. If not, the result is normalised again and returned
+# too
+{ libpath, stringsDir }:
+let
+  lib = import libpath;
+  inherit (lib.path.subpath) normalise;
```

```suggestion
{ lib, strings }:
let
  inherit (lib.path.subpath) normalise;
```

It think the setup should be done outside. See the other scattered comments for concrete suggestions.